src/interchange/avro.rs

```diff
@@ -637,6 +637,17 @@ impl Encoder {
         &self.writer_schema
     }
 
+    pub fn encode_unchecked(&self, schema_id: i32, diff_pair: DiffPair<&Row>) -> Vec<u8> {
+        let mut buf = Vec::new();
+        buf.write_u8(0).expect("writing to vec cannot fail");
+        buf.write_i32::<NetworkEndian>(schema_id)
+            .expect("writing to vec cannot fail");
+        let avro = self.diff_pair_to_avro(diff_pair);
+        debug_assert!(avro.validate(self.writer_schema.top_node()));
+        avro::encode_unchecked(&avro, &self.writer_schema, &mut buf);
+        buf
+    }
+
     /// Encodes a repr::Row to a Avro-compliant Vec<u8>.
     /// See function implementation for Confluent-specific details.
     pub fn encode(&self, schema_id: i32, diff_pair: DiffPair<&Row>) -> Vec<u8> {
```

Comment:
Is there any reason to keep `encode` around? Given that you have the `debug_assert!` I can't think of a single reason you would ever want to use the checked API. (If hypothetically you really need to perform schema validation, e.g., because you're encoding user-supplied Avro values... well, can't you just call validate yourself?)

Reply:
I think it's good to make it clear "hey, you should think about what you're doing" if you do something that is almost guaranteed to cause problems (e.g., decoding an avro value against the wrong schema).

Sort of like how the stdlib has `from_utf8` and `from_utf8_unchecked` -- that is *also* a safety issue, so it's not exactly the same thing, but it's similar enough.

Reply:
Also it wouldn't even surprise me if we force values to always have a reference to a matching schema so you can't even pass them around independently. and then actually make `encode_unchecked` an unsafe function. We're not there yet though.